### PR TITLE
Dom task

### PR DESCRIPTION
**Pull request by kirill8210**
### task name

[Demo](https://kirill8210.github.io/kottans_DOM/) |
[Code base](https://github.com/kirill8210/kottans_DOM)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/kirill8210/dom-task/js/mobile.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const iconMenu = document.querySelector('.icon-menu');
+const menu = document.querySelector('.nav');
+const iconClose = document.querySelector('.icon-close');
+
+const addMenu = () =>{
+    menu.style.left = '0%';
+};
+
+const removeMenu = () =>{
+    menu.style.left = '-100%';
```
Comment (madmaxWMFU) [inline submissions/kirill8210/dom-task/js/mobile.js:10]: Use a methods `Element.classList.add()/remove()/toggle()` of the element.

File: submissions/kirill8210/dom-task/js/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import clubs from './json.js';
+import removeMenu from './mobile.js';
+
+const appClub = document.getElementById('main');
+const appUl = document.getElementById('menu');
```
Comment (madmaxWMFU) [inline submissions/kirill8210/dom-task/js/index.js:5]: Avoid selected elements by ID, better find them by a class name.

File: submissions/kirill8210/dom-task/js/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        <a href="#" class="menu_el">${name}</a>
+    `);
+
+    return clubUl;
+};
+
+const listClub = document.querySelector('.menu');
+
+listClub.addEventListener('click', (e) =>{
+    const target = e.target;
```
Comment (madmaxWMFU) [inline submissions/kirill8210/dom-task/js/index.js:44]:
```suggestion
listClub.addEventListener('click', ({ target }) =>{
```

File: submissions/kirill8210/dom-task/js/index.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+const createUl = (clubs) =>{
+    const { name } = clubs;
+    const clubUl = document.createElement('li');
+
+    clubUl.insertAdjacentHTML('afterbegin', `
+        <a href="#" class="menu_el">${name}</a>
+    `);
+
+    return clubUl;
```
Comment (madmaxWMFU) [inline submissions/kirill8210/dom-task/js/index.js:38]:
```suggestion
    return clubUl.insertAdjacentHTML('afterbegin', `
        <a href="#" class="menu_el">${name}</a>
    `);
```

File: submissions/kirill8210/dom-task/js/index.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        <h2 class="club_title">${name}</h2>
+        <div class="club_about">
+            <div class="club_description">
+                ${description}
+            </div>
+            <div class="club_img"><img src="${img}" alt="${name}"></div>
+        </div>
+    `);
+
+    return clubCards;
```
Comment (madmaxWMFU) [inline submissions/kirill8210/dom-task/js/index.js:27]:
```suggestion
    return clubCards.insertAdjacentHTML('afterbegin', `
        <h2 class="club_title">${name}</h2>
        <div class="club_about">
            <div class="club_description">
                ${description}
            </div>
            <div class="club_img"><img src="${img}" alt="${name}"></div>
        </div>
    `);
```

File: submissions/kirill8210/dom-task/js/index.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+const listClub = document.querySelector('.menu');
+
+listClub.addEventListener('click', (e) =>{
+    const target = e.target;
+    if (target.classList.contains('menu_el')){
+        const clubName = target.textContent;
+        removeMenu();
+        appClub.textContent = '';
+        const clubCards = clubs.filter(club => club.name === clubName).map(createCardClub);
```
Comment (madmaxWMFU) [inline submissions/kirill8210/dom-task/js/index.js:49]: Compare by ID in the filter.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good work. See some comments below.

File: submissions/kirill8210/dom-task/js/mobile.js (hunk 1 above)
Comment (kirill8210) [reply submissions/kirill8210/dom-task/js/mobile.js:10]: ok

File: submissions/kirill8210/dom-task/js/index.js (hunk 2 above)
Comment (kirill8210) [reply submissions/kirill8210/dom-task/js/index.js:5]: ok

File: submissions/kirill8210/dom-task/js/index.js (hunk 3 above)
Comment (kirill8210) [reply submissions/kirill8210/dom-task/js/index.js:44]: ok

File: submissions/kirill8210/dom-task/js/index.js (hunk 6 above)
Comment (kirill8210) [reply submissions/kirill8210/dom-task/js/index.js:49]: ok

File: submissions/kirill8210/dom-task/js/index.js (hunk 4 above)
Comment (kirill8210) [reply submissions/kirill8210/dom-task/js/index.js:38]: Why cut? Your recommendation to shorten the code will call undefined.

File: submissions/kirill8210/dom-task/js/index.js (hunk 5 above)
Comment (kirill8210) [reply submissions/kirill8210/dom-task/js/index.js:27]: Why cut? Your recommendation to shorten the code will call undefined.

**Comment by kirill8210** [comment]
made changes

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Pull request merged**